### PR TITLE
tests: drivers: spi: spi_loopback_test: Add nrf54h20dk support

### PR DESCRIPTION
**Pull request by nordic-krch**
Extend test.

~~PR includes: #71035 and #71036.~~

### CODE REVIEW

**Review by gmarull — DISMISSED** [review]

File: tests/drivers/spi/spi_loopback/boards/nrf54h20dk_nrf54h20_cpuapp.overlay [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	fast@0 {
+		compatible = "test-spi-loopback-fast";
+		reg = <0>;
+		spi-max-frequency = <DT_FREQ_M(8)>;
+	};
+};
+
+&gpio0 {
+	status = "okay";
+};
```
Comment (anangl) [inline tests/drivers/spi/spi_loopback/boards/nrf54h20dk_nrf54h20_cpuapp.overlay:47]: This does not seem to be needed for this test.

Comment (nordic-krch) [reply tests/drivers/spi/spi_loopback/boards/nrf54h20dk_nrf54h20_cpuapp.overlay]: removed.

File: tests/drivers/spi/spi_loopback/boards/nrf54h20dk_nrf54h20_cpuapp.overlay [hunk 2]
```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
```
Comment (anangl) [inline tests/drivers/spi/spi_loopback/boards/nrf54h20dk_nrf54h20_cpuapp.overlay:4]: Incorrect license and copyright date. Sorry that I overlooked this earlier.

**Review by anangl — COMMENTED** [review]
And what about cpurad? Shouldn't we test SPI on it as well? We could actually use a common .dtsi file for this. It could be included by overlays for cpuapp and cpurad that would modify only the `memory-regions` property.

File: tests/drivers/spi/spi_loopback/boards/nrf54h20dk_nrf54h20_cpuapp.overlay (hunk 2 above)
Comment (nordic-krch) [reply tests/drivers/spi/spi_loopback/boards/nrf54h20dk_nrf54h20_cpuapp.overlay]: Fixed.

**Review by anangl — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Review by aescolar — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Pull request merged**